lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -203,16 +218,32 @@ export class PublicClientApplication {
      * Use this function to obtain a token before every call to the API / resource provider
      *
      * MSAL return's a cached token when available
-     * Or it send's a request to the STS to obtain a new token using a hidden iframe.
+     * Or it send's a request to the STS to obtain a new token using a refresh token.
      *
      * @param {@link AuthenticationParameters}
      *
      * To renew idToken, please pass clientId as the only scope in the Authentication Parameters
      * @returns {Promise.<TokenResponse>} - a promise that is fulfilled when this function has completed, or rejected if an error was raised. Returns the {@link AuthResponse} object
      *
      */
-    acquireTokenSilent(request: AuthenticationParameters): Promise<TokenResponse> {
-        throw new Error("Method not implemented."); 
+    async acquireTokenSilent(tokenRequest: TokenRenewParameters): Promise<TokenResponse> {
+        return this.authModule.renewToken(tokenRequest);
+    }
+
+    // #endregion
+
+    // #region Logout
+
+    /**
+     * Use to log out the current user, and redirect the user to the postLogoutRedirectUri.
+     * Default behaviour is to redirect the user to `window.location.href`.
+     */
+    logout(): void {
+        this.authModule.logout().then(logoutUri => {
+            window.location.assign(logoutUri);
```

Comment:
Can we put the code to redirect in a helper function?

Reply:
done.